lib/lib_graph.go

```diff
@@ -107,6 +106,20 @@ func BuildSynchronizedCompleteGraph(n int) ([]Node, Synchronizer) {
 	return vertices, synchronizer
 }
 
+func BuildSynchronizedCompleteGraphWithLoopsAndOrderedNeighbours(n int) ([]Node, Synchronizer) {
+	vertices, synchronizer := BuildSynchronizedEmptyDirectedGraph(n, GetGenerator())
+	chans := getSynchronousChannels(n * n)
+	counter := 0
+	for i := 0; i < n; i++ {
+		for j := 0; j < n; j++ {
+			addOneWayConnection(vertices[i].(*oneWayNode), vertices[j].(*oneWayNode), chans[counter])
```

Comment:
Please explain why you cannot use addTwoWayConnection (the same as in BuildSynchronizedCompleteGraph - only with adding loops and removing shuffleTopology).